### PR TITLE
Team/sun tsu/feature/three strikes

### PR DESCRIPTION
**Pull request by SofijaBosk**
Added a feature in the racing game that gives you a strike that appears on the corner of the screen when a collision happens. The game restarts when collision happens 3 times.

### CODE REVIEW

File: source/dagger/gameplay/racing/racing_game_logic.cpp [hunk 1]
```diff
@@ -20,6 +21,23 @@ void RacingCollisionsLogicSystem::WindDown()
     Engine::Dispatcher().sink<NextFrame>().disconnect<&RacingCollisionsLogicSystem::OnEndOfFrame>(this);
 }
 
+void racing_game::CreateStrike(float TileSize,int life)// Vector3 pos_
+{
+    auto view = Engine::Registry().view<Transform, ControllerMapping, RacingPlayerCar>();
```
Comment (DPrivalov) [inline source/dagger/gameplay/racing/racing_game_logic.cpp]: do we need this view here?

File: source/dagger/gameplay/racing/racing_game_logic.cpp [hunk 2]
```diff
@@ -20,6 +21,23 @@ void RacingCollisionsLogicSystem::WindDown()
     Engine::Dispatcher().sink<NextFrame>().disconnect<&RacingCollisionsLogicSystem::OnEndOfFrame>(this);
 }
 
+void racing_game::CreateStrike(float TileSize,int life)// Vector3 pos_
```
Comment (DPrivalov) [inline source/dagger/gameplay/racing/racing_game_logic.cpp]: parameters should have "_" in the end of the name

File: source/dagger/gameplay/racing/racing_game_logic.cpp [hunk 3]
```diff
@@ -39,7 +57,19 @@ void RacingCollisionsLogicSystem::Run()
...
+                col.colided=false;
+                if(player.pause>0) player.pause--;
+                else
+                {    
+                     
+                    player.pause=200;
+                    player.strike++;
+                    racing_game::CreateStrike(20.f,player.strike);
+                    
+                    if(player.strike==3) m_Restart = true;
```
Comment (DPrivalov) [inline source/dagger/gameplay/racing/racing_game_logic.cpp]: it's better to have habit to use {} everywhere. even in cases like that. 
it makes code more readable

**Review by DPrivalov — CHANGES_REQUESTED** [review]
looks nice!

File: source/dagger/gameplay/racing/racing_game_logic.cpp (hunk 1 above)
Comment (SofijaBosk) [reply source/dagger/gameplay/racing/racing_game_logic.cpp]: No we don't, sorry about that. I was trying something else and forgot to delete it.

**Review by TeodorVasiljevic365 — APPROVED** [review]

**Pull request merged**